Add aws_athena_query_execution and aws_athena_workgroup tables

rinzool opened the pull request:
Related issue: https://github.com/turbot/steampipe-plugin-aws/issues/1664

Note: I created two tables since I needed `aws_athena_workgroup` to build `aws_athena_query_execution`, but I can create two diffrent PRs if necessary :+1: 

# Integration test logs
<details>
  <summary>Logs</summary>

```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results - Table aws_athena_workgroup</summary>

```sql
> select name, effective_engine_version, output_location from aws_athena_workgroup
+-------------+--------------------------+-----------------+
| name        | effective_engine_version | output_location |
+-------------+--------------------------+-----------------+
| workgroup-A | Athena engine version 2  | s3://...        |
| workgroup-B | Athena engine version 3  | s3://...        |
| primary     | Athena engine version 2  | s3://...        |
+-------------+--------------------------+-----------------+

```

</details>


<details>
  <summary>Results - Table aws_athena_query_execution</summary>

```sql
> select workgroup, sum(data_scanned_in_bytes) as total_bytes, avg(engine_execution_time_in_millis) as avg_execution_time from aws_athena_query_execution  group by workgroup
+-------------+----------------+--------------------+
| workgroup   | total_bytes    | avg_execution_time |
+-------------+----------------+--------------------+
| workgroup-A | 24041357049402 | 15437.754816112085 |
| workgroup-B | 23079          | 528                |
| primary     | 36076743       | 831.6390940423437  |
+-------------+----------------+--------------------+

```

</details>

misraved commented:
Thanks a lot @rinzool for creating this PR. I took an initial look at the PR and the changes look good, however, the table docs are slightly lighter in terms of example queries.

Could you please add 2-3 more example queries per table and also resolve the merge conflicts? Thanks 👍.

rinzool commented:
Hi @misraved, I just added a few examples! :+1:

misraved commented:
@rinzool thanks for the new tables 👍, some feedback points on the `aws_athena_query_execution` table:
- Instead of using `workgroup` as the KeyColumn for the ListConfig, I think we can add a `ParentHydrate` that queries all the Athena Workspaces, removing the need for the `workgroup` qual. You could use https://github.com/turbot/steampipe-plugin-aws/blob/main/aws/table_aws_backup_selection.go for reference
- While checking for limits against the maxResults parameter:
```
	if d.QueryContext.Limit != nil {
		limit := int32(*d.QueryContext.Limit)
		if limit < maxResults {
			if limit < 1 {
				maxResults = int32(1)
			} else {
				maxResults = int32(limit)
			}
		}
	}
```

You could replace it with:
```
	if d.QueryContext.Limit != nil {
		limit := int32(*d.QueryContext.Limit)
		if limit < maxResults {
				maxResults = int32(limit)
			}
	  }
```
The SDK will automatically throw an error for any limit input less than 1
- Do we need to use the `TrimSpace()` while checking if the input is empty or not?

rinzool commented:
Thanks again @misraved  for you review :pray:
* I used `PyrentHydrate` it is way better like that :+1: 
* I simplified the condition as you suggested
* I removed the use of `TrimSpace`, i don't think that it's necessary :+1:

misraved reviewed (commented):
Thanks @rinzool for making the suggested changes 👍.

I have added some minor suggestions around the query formatting. In future, you could always use https://www.freeformatter.com/sql-formatter.html#before-output for writing the queries.

To optimize the  ListConfig of `aws_athena_query_execution` table, you could add an optional qual `workgroup` to it.

You could use https://github.com/turbot/steampipe-plugin-aws/blob/main/aws/table_aws_cloudtrail_query.go#L32 and https://github.com/turbot/steampipe-plugin-aws/blob/main/aws/table_aws_cloudtrail_query.go#L156 for reference.

Apologies, I should have suggested this table as a reference for the `ParentHydrate` concept.

rinzool commented:
Hi @misraved I added the optional qual and formatted the queries
However I am not entirely sure that I understood how to manage the optional qual, so don't hesitate to correct me! :+1:

misraved commented on docs/tables/aws_athena_query_execution.md:
````diff
@@ -0,0 +1,60 @@
+
+### Find most used databases
+
+```sql
+
+select
````
```suggestion
```sql
select
```
